pyOCD/gdbserver/gdbserver.py

```diff
@@ -329,12 +338,27 @@ def run(self):
                     break
 
                 if self.packet_io.interrupt_event.isSet():
-                    logging.debug("Got unexpected ctrl-c, ignoring")
+                    if self.non_stop:
```

Comment:
Yeah, that is a good point that all-stop mode would have already halted the target.   I see no downside to the way you have it now, so I wouldn't worry about changing it.  Out of curiosity, how should ctrl-c behave when there are multiple threads?  Should it stop all the threads, or just the current one?  Or should another command such as vCont be used instead of ctrl-c in non-stop mode for well defined behavior among all threads?